framework/helpers/BaseConsole.php

```diff
@@ -957,46 +957,12 @@ public static function startProgress($done, $total, $prefix = '', $width = null)
      */
     public static function updateProgress($done, $total, $prefix = null)
     {
-        $width = self::$_progressWidth;
-        if ($width === false) {
-            $width = 0;
-        } else {
-            $screenSize = static::getScreenSize(true);
-            if ($screenSize === false && $width < 1) {
-                $width = 0;
-            } elseif ($width === null) {
-                $width = $screenSize[0];
-            } elseif ($width > 0 && $width < 1) {
-                $width = floor($screenSize[0] * $width);
-            }
-        }
-        if ($prefix === null) {
-            $prefix = self::$_progressPrefix;
-        } else {
-            self::$_progressPrefix = $prefix;
-        }
-        $width -= static::ansiStrlen($prefix);
-
+        $prefix = $prefix === null ? self::$_progressPrefix : self::$_progressPrefix = $prefix;
```

Comment:
This one-liner is quite complicated with both comparisons and assignments. I'd prefer if-structure instead exactly as it was before.

Additionally, new code does one extra weird assignment: `$prefix = self::$_progressPrefix = $prefix;`.

Reply:
Fixed